flow/layers/image_filter_layer.cc

```diff
@@ -83,23 +85,31 @@ void ImageFilterLayer::Paint(PaintContext& context) const {
   FML_DCHECK(needs_painting(context));
 
   AutoCachePaint cache_paint(context);
-
-  if (layer_raster_cache_item_->IsCacheChildren()) {
-    cache_paint.setImageFilter(transformed_filter_);
-  }
-  if (layer_raster_cache_item_->Draw(context, cache_paint.sk_paint())) {
-    return;
+  if (context.raster_cache) {
+    if (layer_raster_cache_item_->IsCacheChildren()) {
+      cache_paint.setImageFilter(transformed_filter_);
+    }
+    if (layer_raster_cache_item_->Draw(context, cache_paint.sk_paint())) {
+      return;
+    }
   }
 
-  cache_paint.setImageFilter(filter_);
-
-  // Normally a save_layer is sized to the current layer bounds, but in this
-  // case the bounds of the child may not be the same as the filtered version
-  // so we use the bounds of the child container which do not include any
-  // modifications that the filter might apply.
-  Layer::AutoSaveLayer save_layer = Layer::AutoSaveLayer::Create(
-      context, child_paint_bounds(), cache_paint.sk_paint());
-  PaintChildren(context);
+  if (context.leaf_nodes_builder) {
+    cache_paint.setImageFilter(filter_.get());
+    context.leaf_nodes_builder->saveLayer(&child_paint_bounds(),
+                                          cache_paint.dl_paint());
+    PaintChildren(context);
+    context.leaf_nodes_builder->restore();
+  } else {
+    cache_paint.setImageFilter(filter_ ? filter_->skia_object() : nullptr);
```

Comment:
Let's try to standardize with just the Dl variant of this method as mentioned in the ColorFilter version of this PR. I realize that will mean implementing makeWithLocalMatrix which I'll put on a high priority.